### DISCUSSION TITLE
`ScrollSpy` component

### DISCUSSION THREAD

**Discussion by Xelson** [chakra-ui/zag#2136] [Ideas]
# 🚀 Feature request

Add a component to track the currently visible sections like `On this page` block on https://zagjs.com/ or https://ark-ui.com/

## 🧱 Problem Statement / Justification

There is no unified solution for creating components like TOC, the implementation approaches vary and the complexity is quite high: even on the above-mentioned sites, TOC does not work clearly and reliably enough.

## ✅ Proposed solution or API

the component may consist of `ScrollSpy.Root` and `ScrollSpy.Section` parts, we may need more parts to support nesting.
[my react version in ark style](https://gist.github.com/Xelson/ec5a86f8916b035a6eb253346dd21bb1)

## ↩️ Alternatives

[bootstrap](https://getbootstrap.com/docs/5.0/components/scrollspy/)
[mantine TOC](https://github.com/mantinedev/mantine/blob/41b8186aee53aa84d62d3a602f39ac041c10aeef/apps/mantine.dev/src/components/TableOfContents/TableOfContents.tsx#L69)

**Comment by anubra266** [comment]
@Xelson Whipped this up few hours ago:

https://github.com/anubra266/zag-scrollspy
